src/microbe_stage/organelle_unlocks/WorldBasedUnlockCondition.cs

```diff
@@ -0,0 +1,187 @@
+        {
+        }
+    }
+
+    /// <summary>
+    ///   The player species produces more than a specified amount of ATP.
+    /// </summary>
+    public class AtpProductionAbove : WorldBasedUnlockCondition
+    {
+        [JsonProperty]
+        public float Atp;
+
+        public override bool Satisfied()
+        {
+            if (EnergyBalance == null)
+                return false;
+
+            return EnergyBalance.TotalProduction >= Atp;
+        }
+
+        public override void GenerateTooltip(LocalizedStringBuilder builder)
+        {
+            if (EnergyBalance == null)
+                return;
+
+            builder.Append(new LocalizedString("ATP_PRODUCTION_ABOVE", Atp));
+        }
+    }
+
+    /// <summary>
+    ///   The player species produces a specified amount more ATP than it uses.
+    /// </summary>
+    public class ExcessAtpAbove : WorldBasedUnlockCondition
+    {
+        [JsonProperty]
+        public float Atp;
+
+        public override bool Satisfied()
+        {
+            if (EnergyBalance == null)
+                return false;
+
+            return EnergyBalance.FinalBalance >= Atp;
+        }
+
+        public override void GenerateTooltip(LocalizedStringBuilder builder)
+        {
+            if (EnergyBalance == null)
+                return;
+
+            builder.Append(new LocalizedString("EXCESS_ATP_ABOVE", Atp));
+        }
+    }
+
+    /// <summary>
+    ///   The player species has speed that is slower than a certain speed.
+    /// </summary>
+    public class SpeedBelow : WorldBasedUnlockCondition
+    {
+        [JsonProperty]
+        public float Threshold;
+
+        [JsonIgnore]
+        private float speed;
+
+        public override void UpdateData(GameWorld? gameWorld, ICellProperties? playerData,
+            EnergyBalanceInfo? energyBalance)
+        {
+            base.UpdateData(gameWorld, playerData, energyBalance);
+
+            if (PlayerData == null)
+                return;
+
+            var rawSpeed = MicrobeInternalCalculations.CalculateSpeed(PlayerData.Organelles.Organelles,
+                PlayerData.MembraneType,
+                PlayerData.MembraneRigidity,
+                PlayerData.IsBacteria);
+
+            // This needs to be user readable as it is shown by the tooltip
+            speed = (float)Math.Round(MicrobeInternalCalculations.SpeedToUserReadableNumber(rawSpeed), 1);
+        }
+
+        public override bool Satisfied()
+        {
+            return speed < Threshold;
+        }
+
+        public override void GenerateTooltip(LocalizedStringBuilder builder)
+        {
+            builder.Append(new LocalizedString("SPEED_BELOW", Threshold));
+        }
+    }
+
+    /// <summary>
+    ///   The player's current patch has a certain amount of certain compound.
+    /// </summary>
+    public class PatchCompound : WorldBasedUnlockCondition
+    {
+        [JsonProperty("Compound")]
+        public string RawCompound = string.Empty;
```

Comment:
It should also be possible here to just store the Compound, the saving system should be capable of it.